From 5f755f67ce6d930380893b2f83fbd52bf10e17c7 Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Sat, 28 Dec 2019 01:54:30 +0100
Subject: [PATCH 071/122] ASoC: dt-bindings: Add documentation for the Cirrus
 CS4334 stereo codec

The Cirrus CS4334 is a 16-24 bit stereo DAC with I2S input. It does not
have any management interface (like I2C, SPI, GPIOs, etc.).
A functionally compatible replacement is the MXT8234 from Beijing Times
Minxin Technology Co., Ltd.

Add the dt-bindings and a compatible string for the Cirrus and Beijing
Times Minxin Technology Co., Ltd. variant. The latter is just a safety
measure in case any differences to the actual CS4334 codec are found.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 .../bindings/sound/cirrus,cs4334.yaml         | 42 +++++++++++++++++++
 1 file changed, 42 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/sound/cirrus,cs4334.yaml

diff --git a/Documentation/devicetree/bindings/sound/cirrus,cs4334.yaml b/Documentation/devicetree/bindings/sound/cirrus,cs4334.yaml
new file mode 100644
index 000000000000..3941dda1442f
--- /dev/null
+++ b/Documentation/devicetree/bindings/sound/cirrus,cs4334.yaml
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/sound/cirrus,cs4334.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+
+title: Cirrus CS4334 - or functionally compatible - I2S Stereo DACs
+
+maintainers:
+  - Martin Blumenstingl <martin.blumenstingl@googlemail.com>
+
+description: |
+  Cirrus CS4334 16-24 bit stereo DAC with I2S input.
+  https://statics.cirrus.com/pubs/proDatasheet/CS4334-5-8-9_F7.pdf
+  http://www.mxtronics.com/n107/n124/n181/n184/c697/attr/2625.pdf
+
+properties:
+  compatible:
+    enum:
+      - cirrus,cs4334
+      - mxtronics,mxt8234
+
+  "#sound-dai-cells":
+    const: 0
+
+  VDD-supply:
+    description:
+      Regulator for the codec voltage supply
+
+required:
+  - compatible
+  - "#sound-dai-cells"
+  - VDD-supply
+
+examples:
+  - |
+     audio-codec {
+         compatible = "mxtronics,mxt8234";
+         #sound-dai-cells = <0>;
+         VDD-supply = <&codec_vdd>;
+     };
-- 
2.25.1

